In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
sns.set_style('darkgrid')

This data was obtained from [Kaggle](https://www.kaggle.com/datasets/sumansharmadataworld/depression-surveydataset-for-analysis). It is "part of a comprehensive survey aimed at understanding the factors contributing to depression risk among adults." The information was collected from January 2023 - June 2023.

In [8]:
df = pd.read_csv('data/depression_data.csv')
df.tail()

Name  Gender  Age       City Working Professional or Student  \
2551  Raghavendra    Male   25  Bangalore            Working Professional   
2552         Pihu  Female   23       Pune            Working Professional   
2553         Sara  Female   24   Srinagar            Working Professional   
2554       Eshita  Female   56  Bangalore            Working Professional   
2555        Kiran    Male   18      Surat                         Student   

            Profession  Academic Pressure  Work Pressure  CGPA  \
2551        Consultant                NaN            1.0   NaN   
2552           Teacher                NaN            3.0   NaN   
2553        HR Manager                NaN            1.0   NaN   
2554  Business Analyst                NaN            2.0   NaN   
2555               NaN                5.0            NaN   6.7   

      Study Satisfaction  Job Satisfaction     Sleep Duration Dietary Habits  \
2551                 NaN               5.0          5-6 hours        Healthy   
2552                 NaN               1.0  Less than 5 hours       Moderate   
2553                 NaN               4.0  Less than 5 hours       Moderate   
2554                 NaN               3.0          7-8 hours        Healthy   
2555                 3.0               NaN  More than 8 hours      Unhealthy   

        Degree Have you ever had suicidal thoughts ?  Work/Study Hours  \
2551       BBA                                   Yes                12   
2552        MA                                   Yes                 8   
2553        BA                                   Yes                 4   
2554       BBA                                    No                 4   
2555  Class 12                                    No                 6   

      Financial Stress Family History of Mental Illness Depression  
2551                 3                              Yes         No  
2552                 3                               No        Yes  
2553                 4                               No         No  
2554                 5                              Yes         No  
2555                 2                              Yes        Yes

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2556 entries, 0 to 2555
Data columns (total 19 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   Name                                   2556 non-null   object 
 1   Gender                                 2556 non-null   object 
 2   Age                                    2556 non-null   int64  
 3   City                                   2556 non-null   object 
 4   Working Professional or Student        2556 non-null   object 
 5   Profession                             1883 non-null   object 
 6   Academic Pressure                      502 non-null    float64
 7   Work Pressure                          2054 non-null   float64
 8   CGPA                                   502 non-null    float64
 9   Study Satisfaction                     502 non-null    float64
 10  Job Satisfaction                       2054 non-null   float64
 11  Slee

In [7]:
df.isna().sum()

Name                                        0
Gender                                      0
Age                                         0
City                                        0
Working Professional or Student             0
Profession                                673
Academic Pressure                        2054
Work Pressure                             502
CGPA                                     2054
Study Satisfaction                       2054
Job Satisfaction                          502
Sleep Duration                              0
Dietary Habits                              0
Degree                                      0
Have you ever had suicidal thoughts ?       0
Work/Study Hours                            0
Financial Stress                            0
Family History of Mental Illness            0
Depression                                  0
dtype: int64

In [6]:
df.describe()

Age  Academic Pressure  Work Pressure        CGPA  \
count  2556.000000         502.000000    2054.000000  502.000000   
mean     39.043036           3.003984       3.021908    7.567809   
std      12.260596           1.390007       1.417312    1.465367   
min      18.000000           1.000000       1.000000    5.030000   
25%      28.000000           2.000000       2.000000    6.210000   
50%      39.000000           3.000000       3.000000    7.605000   
75%      50.000000           4.000000       4.000000    8.825000   
max      60.000000           5.000000       5.000000   10.000000   

       Study Satisfaction  Job Satisfaction  Work/Study Hours  \
count          502.000000       2054.000000       2556.000000   
mean             3.075697          3.015093          6.023865   
std              1.373490          1.418432          3.771743   
min              1.000000          1.000000          0.000000   
25%              2.000000          2.000000          3.000000   
50%              3.000000          3.000000          6.000000   
75%              4.000000          4.000000          9.000000   
max              5.000000          5.000000         12.000000   

       Financial Stress  
count       2556.000000  
mean           2.968701  
std            1.415527  
min            1.000000  
25%            2.000000  
50%            3.000000  
75%            4.000000  
max            5.000000

* This data contains 2,556 survey responses on 19 metrics. Some of these include age, city, hours of sleep, diet, financial stress, family history of illness, and the target variable, depression.
* There is a mix of string and numeric data. At first glance it looks like we can encode:
    * Working Professional or Student
    * Sleep Duration
    * Dietary Habits
    * Have you ever had suicidal thoughts ?
    * Family History of Mental Illness
    * Depression
* There is null data is several categories which might mean that the respondent did not want to volunteer that information or that the question did not apply to them.
* The average age of respondents is 39.
* Academic and work pressure both average 3 out of 5.
* Similarly, study satisfaction and job satisfaction are 3 out of 5.
* CGPA is a metric that quantifies satisfaction with healthcare services. The range is 1-10 and average response is 7.6.
* Most individuals surveyed work/study for approximately 6 hours per day, while the maximum is 12.
* Individuals of this suvery do not generally experience financial stress, with an average of ~3.

Before jumping into analysis let's standardize the naming convention of the columns to make the data easier to work with as well as encode the columns mentioned above.